ui/desktop/app/routes/scopes/scope/projects/targets/target.js

```diff
@@ -5,21 +5,105 @@
 
 import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { loading } from 'ember-loading';
 
 export default class ScopesScopeProjectsTargetsTargetRoute extends Route {
   // =services
 
   @service store;
+  @service confirm;
+  @service ipc;
+  @service router;
+  @service session;
+
+  queryParams = {
+    isConnecting: {
+      refreshModel: true,
+    },
+  };
 
   // =methods
 
   /**
    * Load a target
    * @param {object} params
    * @param {string} params.target_id
+   * @param {bool} params.isConnecting
    * @return {TargetModel}
    */
-  model({ target_id }) {
-    return this.store.findRecord('target', target_id, { reload: true });
+  async model({ target_id, isConnecting }) {
+    const target = await this.store.findRecord('target', target_id, {
+      reload: true,
+    });
+
+    if (isConnecting) {
+      this.preConnect(target);
+    }
+    return target;
+  }
+
+  /**
+   * Determine if connection workflow based on target attributes.
+   * @param {TargetModel} model
+   */
+  @action
+  @loading
+  async preConnect(target) {
```

Comment:
async is not needed here as we aren't awaiting anything

Reply:
Yep thanks. I originally had more here but removed it and forgot to remove the `async`.

Reply:
Shouldn't we be awaiting the `this.connect`?